backend/de.metas.business/src/main/java/org/adempiere/warehouse/interceptor/C_Order.java

```diff
@@ -0,0 +1,140 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package org.adempiere.warehouse.interceptor;
+
+import de.metas.bpartner.BPartnerContactId;
+import de.metas.bpartner.BPartnerId;
+import de.metas.bpartner.BPartnerLocationId;
+import de.metas.bpartner.exceptions.BPartnerNoBillToAddressException;
+import de.metas.bpartner.service.IBPartnerBL;
+import de.metas.bpartner.service.IBPartnerDAO;
+import de.metas.bpartner.service.IBPartnerOrgBL;
+import de.metas.document.location.DocumentLocation;
+import de.metas.order.location.adapter.OrderDocumentLocationAdapterFactory;
+import de.metas.organization.OrgId;
+import de.metas.user.User;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.adempiere.ad.callout.annotations.Callout;
+import org.adempiere.ad.callout.annotations.CalloutMethod;
+import org.adempiere.ad.callout.spi.IProgramaticCalloutProvider;
+import org.adempiere.ad.modelvalidator.annotations.Interceptor;
+import org.adempiere.ad.modelvalidator.annotations.ModelChange;
+import org.adempiere.warehouse.WarehouseId;
+import org.adempiere.warehouse.api.IWarehouseBL;
+import org.compiere.model.I_C_BPartner;
+import org.compiere.model.I_C_BPartner_Location;
+import org.compiere.model.I_C_Order;
+import org.compiere.model.I_M_Warehouse;
+import org.compiere.model.ModelValidator;
+import org.springframework.stereotype.Component;
+
+import java.util.Optional;
+
+@Callout(I_C_Order.class)
+@Interceptor(I_C_Order.class)
+@Component
+public class C_Order
+{
+	private final IWarehouseBL warehouseBL = Services.get(IWarehouseBL.class);
+	private final IBPartnerOrgBL bPartnerOrgBL = Services.get(IBPartnerOrgBL.class);
+	private final IBPartnerDAO bPartnerDAO = Services.get(IBPartnerDAO.class);
+
+	public C_Order()
+	{
+		Services.get(IProgramaticCalloutProvider.class).registerAnnotatedCallout(this);
+	}
+
+	@CalloutMethod(columnNames = I_C_Order.COLUMNNAME_M_Warehouse_ID)
+	@ModelChange(timings = ModelValidator.TYPE_BEFORE_CHANGE, ifColumnsChanged = I_C_Order.COLUMNNAME_M_Warehouse_ID)
+	public void updateBillPartnerAndDropshipFlagFromWarehouse(@NonNull final I_C_Order order)
+	{
+		if (order.getM_Warehouse_ID() <= 0 || order.isSOTrx())
+		{
+			return; // nothing to do
+		}
+
+		final WarehouseId warehouseId = WarehouseId.ofRepoId(order.getM_Warehouse_ID());
+		final OrgId orgId = OrgId.ofRepoId(order.getAD_Org_ID());
+
+		final DocumentLocation billToAddress = getBillingDocumentForExternalBPartner(warehouseId, orgId)
+				.orElseGet(() -> getBillingDocumentForBPartnerId(BPartnerId.ofRepoId(order.getC_BPartner_ID())));
```

Comment:
i think if `getBillingDocumentForExternalBPartner(warehouseId, orgId)` doesn't give us a location, three is no point in falling back to order.getC_BPartner_ID()'s location..

Reply:
@metas-ts we thought about setting them back to order.getC_BPartner_ID in case the user changes it's mind and goes back to a warehouse which isn't configured with the newly added flag

Reply:
@mi-ps ok, but then please see my comment further down: https://github.com/metasfresh/metasfresh/pull/13498#discussion_r955069371 (DRY)